### ISSUE TITLE
Warning should be issued for full word matches only.

### ISSUE THREAD

**Issue by merill** [microsoft/InclusivenessAnalyzer#15]
In the example below a warning is raised for the term 'he' against the word 'heart'.

This is a false flag. The scan should be for the whole word.

In the VS plug-in we ignore matches if the word length is less than five characters. We could do the same here or use an alternate approach. Ideally it would be good to keep the logic in sync.

```c#
        /// <summary>
        /// Checks if term was found in the symbol.
        /// Match whole word if it has less than 5 characters
        /// </summary>
        /// <param name="symbol">The phrase that is being checked.</param>
        /// <param name="term">The non-inclusive phrase we are looking for.</param>
        /// <returns></returns>
        private static bool IsMatch(string symbol, string term)
        {
            return term.Length < 5 ?
                symbol.Equals(term, StringComparison.InvariantCultureIgnoreCase) :
                symbol.IndexOf(term, StringComparison.InvariantCultureIgnoreCase) >= 0;
            
        }
```
Source: https://github.com/microsoft/InclusivenessAnalyzerVisualStudio/blob/main/src/InclusivenessAnalyzer/InclusivenessAnalyzer/InclusivenessAnalyzer.cs#L139

```
Warning: Found the term 'he', consider using alternatives: they
  Warning: 	[Line 4]     description: 'Fails the step if a non-inclusive term is found'
  Warning: 	[Line 1[8](https://github.com/microsoft/InclusivenessAnalyzer/actions/runs/3086984421/jobs/4991883279#step:5:9)]   icon: 'heart'
  Warning: Found the term 'his', consider using alternatives: their,theirs
```

**Comment by cblomart** [comment]
This has been implemented in latest PR #21
